### DISCUSSION TITLE
Rule for validating CSS syntax when required?

### DISCUSSION THREAD

**Discussion by schwigri** [eslint/eslint#15013] [Ideas]
Hi, not sure if a rule like this exists or if one can, but for APIs that require CSS syntax (such as `document.querySelector()`), I would like to get a warning if the argument is not escaped.

For example, I am setting element IDs via UUID, which can occasionally begin with a digit. While this is valid HTML, it is invalid CSS unless escaped. See the following example:

    const elId = getElId(); /* this function returns the UUID, which could be "1234asdf" */
    const target = document.querySelector(`#${elId} .target`);

In this case, it will throw an error since the argument is invalid CSS syntax. The solution would be something like this:

    const elId = getElId();
    const target = document.querySelector(`#${CSS.escape(elId)} .target`);

I would also like to see a warning in case that a hard-coded invalid string is used, just to prevent any unexpected runtime errors. For example, `document.querySelector("#1234asdf .target");` should give warning as well.

If a rule for this exists, I would greatly appreciate being pointed in the right direction!

**Comment by nzakas** [comment]
There isn’t a core rule that does this. Your best bet is to search for a plugin.

**Comment by bmish** [comment]
We have a rule in [eslint-plugin-ember](https://github.com/ember-cli/eslint-plugin-ember) that is related to this: [ember/require-valid-css-selector-in-test-helpers](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/require-valid-css-selector-in-test-helpers.md).

**Reply by schwigri** [reply]
This is really helpful, thank you for sharing!
